### ISSUE TITLE
Support Text to Speech

### ISSUE THREAD

**Issue by zolrath** [elixir-nx/bumblebee#209]
Hello!
As Speech to Text models such as Whisper are added having access to some of the impressive AI Text to Speech models would be a nice way to close the loop!

My current suggestion for a model to support would be [bark](https://github.com/suno-ai/bark).

**Comment by fredwu** [comment]
+1

Would also love to see the support for [Coqui TTS](https://github.com/coqui-ai/TTS).

**Comment by bartekupartek** [comment]
It would be great to run Bark in Elixir, also recently this TTS model brought a lot of attention [https://github.com/collabora/WhisperSpeech](WhisperSpeech)

**Comment by Jdyn** [comment]
I hate to reiterate what's already been said but TTS in Bumblebee using Bark would be super valuable. Any chance of supporting it?

Hugging face: https://huggingface.co/suno/bark

**Comment by josevalim** [comment]
Pull requests are always welcome. Starting with one of the models in Hugging Face Transformers is probably the easiest way to get started: https://huggingface.co/docs/transformers/en/tasks/text-to-speech

**Comment by nickkaltner** [comment]
Just adding this as an interesting model to support too https://huggingface.co/coqui/XTTS-v2

**Comment by bartekupartek** [comment]
I tried to port Bark and later on WhisperSpeech, they use multiple models to convert text to semantics, semantics to audio and and encode... anyway there are more promising models recently released https://github.com/jasonppy/VoiceCraft
or https://github.com/myshell-ai/OpenVoice
After reviewing their architectures they might be easier to integrate

**Comment by michelson** [comment]
@bartekupartek, do you have your implementation open?  I'm trying to do the same I've read the docs but not sure where to start.

**Comment by bartekupartek** [comment]
@michelson not yet but working on it, this models aren't using standard layers, I needed to move back to understand simpler models with axon first

**Comment by josevalim** [comment]
Correct. We would need to implement them in Elixir. Maybe @polvalente knows of an implementation that could be ported, otherwise we need to look if there are any Jax implementations. If not, maybe it needs to be a separate library we invoke.

**Comment by polvalente** [comment]
There are many kinds of vocoders. I think the best way to approach this would be to choose a specific model we want to support and work towards porting the one it uses.

**Comment by bartekupartek** [comment]
I was thinking we might implement one of pytorchaudio vocoders like Griffin-Lim(sounds robotic) or WaveRNN or Nvidia Waveglow to turn mel spectograms into audio, but I just read trough whole VALL-E paper Bark is based on: 

> We propose VALL-E, the first TTS framework with strong in-context learning capabilities as
GPT-3, which treats TTS as a language model task with audio codec codes as an intermediate
representation to replace the traditional mel spectrogram

It would be fun to have Tacotron 2 working end to end but it looks like it doesn't make sense for any recent models mentioned above that are using [facebook/encodec](https://huggingface.co/facebook/encodec_24khz) to turn outputs into audio codes directly :bowing_man:

**Comment by username14415** [comment]
+1

**Comment by louwers** [comment]
Shouldn't this be closed? https://hexdocs.pm/bumblebee/Bumblebee.Audio.html

**Comment by polvalente** [comment]
@louwers The issue title text to speech, that module is speech to text (the opposite direction)

**Comment by louwers** [comment]
In my defense, it was late.

Thanks for clarifying.
